File: app/src/main/java/com/imangazalievm/bubbble/di/ShotsPresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.ShotsPresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.ShotsPresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = ShotsPresenterModule.class)
public interface ShotsPresenterComponent {

    ShotsPresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/shotslist/ShotsPresenterComponent.kt
package com.imangazalievm.bubbble.di.shotslist

import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.presentation.shotslist.ShotsPresenter
import dagger.Component

@Presenter
@Component(dependencies = [ApplicationComponent::class], modules = [ShotsPresenterModule::class])
interface ShotsPresenterComponent {

    fun getPresenter(): ShotsPresenter

}